### PR TITLE
[CARBONDATA-2519] Add document for CarbonReader

### PR DESCRIPTION
**Pull request by xubo245**
Add document for CarbonReader
Be sure to do all of the following checklist to help us incorporate 
your contribution quickly and easily:

 - [x] Any interfaces changed?
 No
 - [x] Any backward compatibility impacted?
 No
 - [x] Document update required?
Yes
 - [x] Testing done
        Please provide details on 
        - Whether new unit test cases have been added or why no new tests are required?
        - How it is tested? Please attach test report.
        - Is it a performance related change? Please attach the performance test report.
        - Any additional information to help reviewers in testing this change.
       No need
 - [x] For large changes, please consider breaking it into sub-tasks under an umbrella JIRA. 
No

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/6075/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/4915/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/5080/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/6078/

**Comment by xubo245** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/6083/

**Comment by xubo245** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/4924/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/6085/

File: store/sdk/src/main/java/org/apache/carbondata/sdk/file/CarbonReaderBuilder.java [hunk 1]
```diff
@@ -136,10 +160,17 @@ public CarbonReaderBuilder setEndPoint(String key, String value) {
    * @return CarbonWriterBuilder
    */
   public CarbonReaderBuilder setEndPoint(String value) {
-    FileFactory.getConfiguration().set(Constants.ENDPOINT, value);
-    return this;
+    return setEndPoint(Constants.ENDPOINT, value);
```
Comment (ajantha-bhat) [inline store/sdk/src/main/java/org/apache/carbondata/sdk/file/CarbonReaderBuilder.java]: There is one more implementation of setEndPoint. Please add that also

File: docs/sdk-guide.md [hunk 2]
```diff
@@ -398,3 +399,160 @@ Reference : [list of carbon properties](http://carbondata.apache.org/configurati
...
+
+```
+ /**
+   * Set the endpoint for S3
+   *
+   * @param key   the string of endpoint for different S3 type,like: fs.s3a.endpoint
+   * @param value the value of endpoint
+   * @return CarbonWriterBuilder
+   */
+  public CarbonReaderBuilder setEndPoint(String key, String value);
```
Comment (ajantha-bhat) [inline docs/sdk-guide.md]: There is one more implementation of setEndPoint. Please add that also

File: docs/sdk-guide.md [hunk 3]
```diff
@@ -398,3 +399,160 @@ Reference : [list of carbon properties](http://carbondata.apache.org/configurati
...
+```
+
+```
+  /**
+   * Close reader
+   */
+  public void close();
+```
+
+###org.apache.carbondata.sdk.file.CarbonReaderBuilder
```
Comment (ajantha-bhat) [inline docs/sdk-guide.md]: need to give space after ###, else formatting will not reflect

File: docs/sdk-guide.md [hunk 4]
```diff
@@ -398,3 +399,160 @@ Reference : [list of carbon properties](http://carbondata.apache.org/configurati
...
+    
+    // 3. Close this reader
+    reader.close();
+```
+
+Find example code at [CarbonReaderExample](https://github.com/apache/carbondata/blob/master/examples/spark2/src/main/java/org/apache/carbondata/examples/sdk/CarbonReaderExample.java) in the CarbonData repo.
+
+## API List
+
+### org.apache.carbondata.sdk.file.CarbonReader
```
Comment (ajantha-bhat) [inline docs/sdk-guide.md]: change it to below. Just to be same as above style

### org.apache.carbondata.sdk.file.CarbonReader

File: docs/sdk-guide.md (hunk 2 above)
Comment (xubo245) [reply docs/sdk-guide.md]: ok, AK and SK also have another interface

File: docs/sdk-guide.md (hunk 3 above)
Comment (xubo245) [reply docs/sdk-guide.md]: ok, done

File: docs/sdk-guide.md (hunk 4 above)
Comment (xubo245) [reply docs/sdk-guide.md]: ok, added

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/6102/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/4940/

**Comment by xubo245** [comment]
@ajantha-bhat CI pass, please review it

File: store/sdk/src/main/java/org/apache/carbondata/sdk/file/CarbonReaderBuilder.java [hunk 5]
```diff
@@ -50,26 +50,50 @@
...
   public CarbonReaderBuilder projection(String[] projectionColumnNames) {
     Objects.requireNonNull(projectionColumnNames);
     this.projectionColumns = projectionColumnNames;
     return this;
   }
 
+  /**
+   * Configure the transactional status of table
+   *
+   * @param isTransactionalTable whether is transactional table or not
```
Comment (ajantha-bhat) [inline store/sdk/src/main/java/org/apache/carbondata/sdk/file/CarbonReaderBuilder.java]: update the default value and mention that if false, read from flat folder. 

Else user will not know what is this API for.

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/5096/

File: store/sdk/src/main/java/org/apache/carbondata/sdk/file/CarbonReaderBuilder.java (hunk 5 above)
Comment (xubo245) [reply store/sdk/src/main/java/org/apache/carbondata/sdk/file/CarbonReaderBuilder.java]: ok, Done

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/5102/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/6108/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/4946/

**Comment by xubo245** [comment]
@ajantha-bhat CI pass， pease check

**Comment by ajantha-bhat** [comment]
LGTM

**Comment by kunal642** [comment]
LGTM

**Pull request closed without merging**